atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -206,24 +212,25 @@ private synchronized void updateStores(TransactionsLockWatchUpdate updateForTran
         Multimap<Sequence, StartTimestamp> reversedMap = createSequenceTimestampMultimap(updateForTransactions);
 
         // Without this block, updates with no events would not store a snapshot.
-        currentVersion.map(LockWatchVersion::version).map(Sequence::of).ifPresent(sequence -> {
+        Optional<Sequence> maybeSeq =
+                currentVersion.map(LockWatchVersion::version).map(Sequence::of);
+        if (maybeSeq.isPresent()) {
             synchronized (this) {
-                snapshotStore.storeSnapshot(sequence, reversedMap.get(sequence), valueStore.getSnapshot());
+                snapshotStore.storeSnapshot(maybeSeq.get(), reversedMap.get(maybeSeq.get()), valueStore.getSnapshot());
             }
-        });
+        }
 
-        updateForTransactions.events().stream().filter(this::isNewEvent).forEach(event -> {
-            synchronized (this) {
+        for (LockWatchEvent event : updateForTransactions.events()) {
+            if (isNewEvent(event)) {
                 valueStore.applyEvent(event);
                 Sequence sequence = Sequence.of(event.sequence());
                 snapshotStore.storeSnapshot(sequence, reversedMap.get(sequence), valueStore.getSnapshot());
             }
-        });
+        }
 
-        updateForTransactions
-                .startTsToSequence()
-                .keySet()
-                .forEach(timestamp -> cacheStore.createCache(StartTimestamp.of(timestamp)));
+        for (Long timestamp : updateForTransactions.startTsToSequence().keySet()) {
+            cacheStore.createCache(StartTimestamp.of(timestamp));
+        }
```

Comment:
Was the `@GuardedBy` check complaining here as well? We're only guarding the value and snapshot stores, so that'd be odd. Or is this a perf improvement? Either way, we can probably not use boxed longs for the for loop, no (unless there is a perf overhead in unboxing, since presumably the keyset is of boxed longs...)

Reply:
was over-eager with unwinding streams, switching back